internal/cmd/gocdk/init_test.go

```diff
@@ -147,6 +148,69 @@ func TestInit(t *testing.T) {
 		}
 	})
 }
+func TestInferModulePath(t *testing.T) {
+	testCases := []struct {
+		name       string
+		testGOPATH func(string) string
+		wantErr    bool
+	}{
+		{
+			"no GOPATH entry",
+			func(dir string) string {
+				return ""
+			},
+			true,
+		},
+		{
+			"single GOPATH entry",
+			func(dir string) string {
+				return "GOPATH=" + dir
+			},
+			false,
+		},
+		{
+			"multiple GOPATH entries",
+			func(dir string) string {
+				multiPath := "GOPATH=" + dir + ":" + filepath.Join("", "some", "other", "path")
```

Comment:
Use `filepath.ListSeparator` to keep portable:

```suggestion
				multiPath := "GOPATH=" + dir + string(filepath.ListSeparator) + filepath.Join("", "some", "other", "path")
```